sdk/core/Azure.Core/tests/samples/OperationSamples.cs

```diff
@@ -51,6 +52,33 @@ public async Task OperationUpdateStatus()
             #endregion
         }
 
+        [Test]
+        [Ignore("Only verifying that the sample builds")]
+        public async Task OperationCustomPollingCompletion()
+        {
+            // create a client
+            SecretClient client = new SecretClient(new Uri("http://example.com"), new DefaultAzureCredential());
+
+            #region Snippet:OperationCustomPollingCompletion
+            // Start the operation
+            DeleteSecretOperation operation = await client.StartDeleteSecretAsync("SecretName");
+
+            while (true)
+            {
+                // there could be more complex logic to control polling interval
+                Thread.Sleep(TimeSpan.FromSeconds(60));
+
+                Response state = await operation.UpdateStatusAsync();
+                Console.WriteLine($"Operation status is {state}");
+
+                if (operation.HasCompleted)
+                {
+                    break;
+                }
+            }
```

Comment:
I think we can simplify the loop. Also, I think it may be better not to try and print the response to the console as it would n't provide any information about the response content.

```suggestion
            while (operation.HasCompleted)
            {
                // there could be more complex logic to control polling interval
                Thread.Sleep(TimeSpan.FromSeconds(60));

                Response statusResponse = await operation.UpdateStatusAsync();
                // optionally do something with the statusResponse
            }
```